API-Samples/Java/app/src/main/java/apisamples/VideoIndexerClient.java

```diff
@@ -101,7 +108,15 @@ public String uploadVideo(String videoUrl, String videoName) {
         map.put("privacy", "private");
         map.put("partition", "partition");
         map.put("videoUrl", videoUrl);
+        // For API Based Scenarios it is advised to set "NoStream" for faster indexing. 
         map.put("streamingPreset","NoStreaming");
+        //Retention Period of Video in days. Default is No retention. Max Allowed value is 7.
+        map.put("retentionPeriod","1");
+        // Use Callback URL to get notified on Video Indexing Events ( Start/ End Processing)
+        if (!CallbackUrl.isBlank()) {
+            map.put("callbackUrl", CallbackUrl);

```

Comment:
Tomer, I think that the value should be encoded 
String encodedUrl = URLEncoder.encode(originalUrl, StandardCharsets.UTF_8);

This is also true for the videoUrl param to be honest.

Reply:
Totally Agree ! 
thanks .